Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1029,29 +1031,39 @@ The decoded 16-character input string totals 10 bytes, so you know that the vali
 ╰           0x00001868      pop {r4, fp, pc}                           ; entry.preinit0 ; entry.preinit0 ;
 ```
 
-If you look in the graph view you can see a loop with some XOR-magic happening at 0x00001784, which supposedly decodes the input string.
+Discussing all the instructions in the function is beyond the scope of this chapter, instead we will discuss only the important points need for the analysis. In the validation function, there is a loop present at `0x00001784`, and inside the loop there is XOR operation being performed.
 
-![Loop](Images/Chapters/0x05c/loop_1784.png)
+> 0x00001798      eor r3, r2, r3
 
-Starting from 0x000017dc, you can see a series of decoded values compared with values from further subfunction calls.
+The loop is more clearly visible in the graph view below.
+
+![XOR instruction inside a loop](Images/Chapters/0x05c/loop_1784.png)
+
+A note on XOR encryption, XOR is a very commonly used technique to encrypt information where obfuscation is the goal rather than security. XOR should not be used for any serious encryption, as it can be cracked using frequency analysis. Presence of XOR in such validation logic always require special attention and analysis.
```

Comment:
```suggestion
A note on XOR encryption: XOR is a very commonly used technique to encrypt information where obfuscation is the goal rather than security. XOR should not be used for any serious encryption, as it can be cracked using frequency analysis. Presence of XOR in such validation logic always requires special attention and analysis.
```